File: Dart/clases.dart

class Carro {
 double?  peso; 
 String? marca;
 String? modelo;
 double? tamanio;  

 Carro ({required this.peso, required this.marca, required this.modelo, required this.tamanio});
 @override
  String toString() {
    return 'Carro $peso, $marca, $modelo,  $tamanio';
  }
}
class Celular {
 double? peso; 
 String? color;
 String? marca;

 Celular({required this.peso, required this.color, this.marca}); 
 @override
  String toString() {
    return 'Celular $peso, $marca, $color';  
  }
}
class Ventana {
  int? tamanio;
  int? ancho;
  String? color;

  Ventana({required this.tamanio, required this.ancho, required this.color}); 

  @override
  String toString() {
    return 'Ventana $tamanio, $ancho, $color';
  }
}

void main () {
  Carro carro = new Carro(
    peso: 1.200,
    marca: "toyota",
    modelo: "hilux",
    tamanio: 3.0,
  );
  Celular celular = new Celular( 
    peso: 2.0,
    color: "rojo",
    marca: "huawei"
  );

  Ventana ventana = new Ventana(tamanio: 20, ancho: 10, color: "azul");
  
  print(celular);
  print(carro);
  print(ventana);
} 

File: Node/clases.js
class Celular{
	constructor(modelo,marca,color){
		this.modelo = modelo;
		this.marca = marca;
		this.color = color;
	}

	funcionar () {
		console.log(`El celular de marca ${this.marca} modelo ${this.modelo} con color ${this.color} funciona correctamente`);
	}

	dañado () {
		console.log(`El celular de la marca ${this.marca} modelo ${this.modelo} con color ${this.color} se encuentra dañado por favor revisar con su tecnico de confianza`);
	}
}

// hacer instancias de la clase
const celularAlto = new Celular("Smart", "Infinix", "titanio");
const celularBajo = new Celular("Huawei", "y9", "negro");


// acceder a un metodo
console.log(celularAlto.modelo);
console.log(celularBajo.marca);


// Imprimir usando el modelo
celularAlto.funcionar();
console.log(celularAlto.funcionar);

celularBajo.dañado();
console.log(celularBajo.dañado);

